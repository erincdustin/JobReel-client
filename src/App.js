import React from 'react';
import logo from './logo.svg';
import { Route } from 'react-router';
import Landing from './routes/Landing'
import './App.css';

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Route 
          exact path="/"
          component={Landing}
        />
      </div>
    );
  }
}

export default App;
