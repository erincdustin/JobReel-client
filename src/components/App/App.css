.App {
  margin: 0 auto;
  max-width: 100vw;
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
  min-width: 320px;
  height: 100vh;
  min-height: 100vh;
}

.header {
  grid-column-start: 1;
  grid-column-end: 13;
}


main {
  grid-column-start: 3;
  grid-column-end: 10;
}

