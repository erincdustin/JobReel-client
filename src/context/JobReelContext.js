import React, { Component } from 'react'
import AuthApiService from '../services/auth-api-service'
import TokenService from '../services/token-service'
import IdleService from '../services/idle-service'

const JobReelContext = React.createContext({
    error: null,
    user: {}, //nested Resume object assigned to User?
    search: {},
    jobData: {},
    savedJobs: [],
    savedEvents: [],
    authenticJobs: [],
    gitHubJobs: [],
    companies: [],
    resources: [],
    contacts: [],
    meetups: [],
    professionals: [],
    manualJobAdd: false,
    manualEventAdd: false,
    manualCompanyAdd: false,
    setError: () => { },
    clearError: () => { },
    processLogin: () => { },
    processLogout: () => { },
    setUser: () => { },
    setAuthenticJobs: () => [],
    setGithubJobs: () => [],
    setCompanies: () => { },
    setResources: () => { },
    setContacts: () => { },
    setMeetups: () => { },
    setProfessionals: () => { },
    setJobData: () => { },
    setSearch: () => { },
    setManualJobAdd: () => { },
    setManualEventAdd: () => { },
    setManualCompanyAdd: () => { },
    setSavedJobs: () => { },
    deleteJob: () => { },
    updateJob: () => { },
    updateEvent: () => { },
    deleteEvent: () => { },
    updateCompany: () => { },
    deleteCompany: () => { },
})

export default JobReelContext

export class JobReelProvider extends Component {
    constructor(props) {
        super(props)
        const state = {
            error: null,
            user: {}, //nested Resume object assigned to User?
            search: {},
            authenticJobs: [],
            gitHubJobs: [],
            companies: [],
            resources: [],
            contacts: [],
            meetups: [],
            professionals: [],
            savedJobs: [],
            savedEvents: [],
            jobData: {},
            manualJobAdd: false,
            manualEventAdd: false,
            manualCompanyAdd: false,
            setError: this.setError,
            clearError: this.clearError,
            setUser: this.setUser,
            setAuthenticJobs: this.setAuthenticJobs,
            setGithubJobs: this.setGithubJobs,
            setSavedJobs: this.setSavedJobs,
            setSavedEvents: this.setSavedEvents,
            setCompanies: this.setCompanies,
            setResources: this.setResources,
            setContacts: this.setContacts,
            setMeetups: this.setMeetups,
            setProfessionals: this.setProfessionals,
            setJobDetails: this.setJobDetails,
            setJobStatus: this.setJobStatus,
            processLogin: this.processLogin,
            processLogout: this.processLogout,
            handleSubmit: this.handleSubmit,
            setJobData: this.setJobData,
            setSearch: this.setSearch,
            setManualJobAdd: this.setManualJobAdd,
            setManualEventAdd: this.setManualEventAdd,
            setManualCompanyAdd: this.setManualCompanyAdd,
            deleteJob: this.deleteJob,
            deleteEvent: this.deleteEvent,
            updateJob: this.updateJob,
            updateEvent: this.updateEvent,
            updateCompany: this.updateCompany,
            deleteCompany: this.deleteCompany,
        }

        const jwtPayload = TokenService.parseAuthToken()

        if (jwtPayload)
            state.user = {
                id: jwtPayload.user_id,
                name: jwtPayload.name,
                username: jwtPayload.sub,
            }

        this.state = state;
        IdleService.setIdleCallback(this.logoutBecauseIdle)
    }

    componentDidMount() {
        if (TokenService.hasAuthToken()) {
            IdleService.registerIdleTimerResets()
            TokenService.queueCallbackBeforeExpiry(() => {
                this.fetchRefreshToken()
            })
        }
    }

    componentWillUnmount() {
        IdleService.unRegisterIdleResets()
        TokenService.clearCallbackBeforeExpiry()
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setUser = user => {
        this.setState({ user })
    }

    setUserMeetupProfile = (user, accesstoken) => {
        console.log(user, accesstoken)
        user.meetupProfile = accesstoken
    }

    setJobData = jobData => {
        this.setState({
            jobData
        })
        this.setJobs(jobData.listing)
    }

    setAuthenticJobs = authenticJobs => {
        this.setState({ authenticJobs })
    }

    setGithubJobs = gitHubJobs => {
        this.setState({ gitHubJobs })
    }

    setSavedJobs = savedJobs => {
        this.setState({ savedJobs })
    }

    setSavedEvents = savedEvents => {
        this.setState({ savedEvents })
    }

    setCompanies = companies => {
        this.setState({ companies })
    }

    setResources = resources => {
        this.setState({ resources })
    }

    setContacts = contacts => {
        this.setState({ contacts })
    }

    setMeetups = meetups => {
        this.setState({ meetups })
    }

    setProfessionals = professionals => {
        this.setState({ professionals })
    }

    setSearch = search => {
        const location = search.location
        const jobTitle = search.jobTitle
        this.setState({search : {location, jobTitle}})
    }

    setManualJobAdd = status => {
        this.setState({ manualJobAdd: status })
    }

    setManualEventAdd = status => {
        this.setState({ manualEventAdd: status })
    }

    setManualCompanyAdd = status => {
        this.setState({ manualCompanyAdd: status })
    }

    //INDEED API METHOD
    // setJobDetails = (details, jobkey) => {

    //     const updatedJobs = [...this.state.jobs]
    //     updatedJobs.find(job => job.jobkey === jobkey).details = details;
    //     this.setState({
    //         jobs: updatedJobs
    //     })
    // }

    setJobStatus = (status, jobs, jobkey) => {
        const updatedJobs = [...jobs]
        updatedJobs.find(job => job.jobkey === jobkey).status = status;
        this.setState({
            jobs: updatedJobs
        })
    }

    deleteJob = jobId => {
        this.setState({
            savedJobs: this.state.savedJobs.filter(job => job.job_id !== jobId)
        });
    }

    deleteEvent = eventId => {
        this.setState({
            savedEvents: this.state.savedEvents.filter(event => event.event_id !== eventId)
        });
    }

    updateJob = (updatedJob) => {
        this.setState({
            savedJobs: this.state.savedJobs.map(job => 
               (job.job_id !== updatedJob.job_id) ? job : updatedJob 
            )
        })
    }

    updateEvent = (updatedEvent) => {
        this.setState({
            savedEvents: this.state.savedEvents.map(event => 
               (event.event_id !== updatedEvent.event_id) ? event : updatedEvent 
            )
        })
    }

    deleteCompany = companyId => {
        this.setState({
            companies: this.state.companies.filter(company => company.company_id !== companyId)
        });
    }

    updateCompany = (updatedCompany) => {
        this.setState({
            companies: this.state.companies.map(company => 
               (company.company_id !== updatedCompany.company_id) ? company : updatedCompany 
            )
        })
    }

    updateEvent = (updatedEvent) => {
        this.setState({
            savedEvents: this.state.savedEvents.map(event => 
               (event.event_id !== updatedEvent.event_id) ? event : updatedEvent 
            )
        })
    }

    processLogin = authToken => {
        TokenService.saveAuthToken(authToken)
        const jwtPayload = TokenService.parseAuthToken()
        this.setUser({
            id: jwtPayload.user_id,
            name: jwtPayload.name,
            username: jwtPayload.sub,
        })
        IdleService.registerIdleTimerResets()
        TokenService.queueCallbackBeforeExpiry(() => {
            this.fetchRefreshToken()
        })
    }

    processLogout = () => {
        TokenService.clearAuthToken()
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unRegisterIdleResets()
        this.setUser({})
    }

    logoutBecauseIdle = () => {
        TokenService.clearAuthToken()
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unRegisterIdleResets()
        this.setUser({ idle: true })
    }

    fetchRefreshToken = () => {
        AuthApiService.refreshToken()
            .then(res => {
                TokenService.saveAuthToken(res.authToken)
                TokenService.queueCallbackBeforeExpiry(() => {
                    this.fetchRefreshToken()
                })
            })
            .catch(err => {
                this.setError(err)
            })
        }

    render() {
        return (
            <JobReelContext.Provider value={this.state}>
                {this.props.children}
            </JobReelContext.Provider>
        )
    }
}
