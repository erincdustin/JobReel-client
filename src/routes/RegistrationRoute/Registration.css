.registration-page {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
  background-attachment: scroll;
  position: relative;
}

.registration-image > img {
  float: left;
  height: 320px;
  width: 320px;
  position: absolute;
  left: -35%;
  z-index: -1;
}

.form-registration {
  grid-column-start: 6;
  grid-column-end: 9;
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  height: 370px;
  background: #3b3a50;
  color: #eeebd9;
  box-shadow: 0.5em 0.5em 0.2em rgb(48, 39, 39);
}

.registration-description {
  grid-column-start: 1;
  grid-column-end: 5;
  height: 25em;
}

.error {
  color: red;
}

@media screen and (max-width: 960px) {
  .form-registration {
    grid-column-start: 2;
    grid-column-end: 8;
  }
}