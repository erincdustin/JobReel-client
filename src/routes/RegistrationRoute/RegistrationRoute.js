import React, { Component } from 'react'
import Description from '../../components/Description/Description'
import { Link } from 'react-router-dom'
import { Input, Required, Label } from '../../components/Form/Form'
import AuthApiService from '../../services/auth-api-service'
import Button from '../../components/Button/Button'



export default class RegistrationRoute extends Component {
    static defaultProps = {
        history: {
            push: () => { },
        },
    }

    state = { error: null }

    firstInput = React.createRef()

    handleSubmit = ev => {
        ev.preventDefault()
        const { email, first_name, last_name, username, password } = ev.target
        AuthApiService.postUser({
            email: email.value,
            first_name: first_name.value,
            last_name: last_name.value,
            username: username.value,
            password: password.value,
        })
            .then(() => {
                email.value = ''
                first_name.value = ''
                last_name.value = ''
                username.value = ''
                password.value = ''
                const { history } = this.props
                history.push('/login')
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
    }

    componentDidMount() {
        this.firstInput.current.focus()
    }

    render() {
        const { error } = this.state
        return (
            <section className="authSection">
                <form id="form-container"
                    onSubmit={this.handleSubmit}
                >
                    <div role='alert'>
                        {error && <p>{error}</p>}
                    </div>
                    <div>
                        <Label htmlFor='registration-email-input'>
                            Email:<Required />
                        </Label>
                        <Input
                            ref={this.firstInput}
                            id='registration-email-input'
                            name='email'
                            required
                        />
                    </div>
                    <div>
                        <Label htmlFor='registration-first-name-input'>
                            First name:<Required />
                        </Label>
                        <Input

                            id='registration-first-name-input'
                            name='first-name'
                            required
                        />
                    </div>
                    <div>
                        <Label htmlFor='registration-last-name-input'>
                            Last name:<Required />
                        </Label>
                        <Input
                            id='registration-last-name-input'
                            name='last-name'
                            required
                        />
                    </div>
                <div>
                    <Label htmlFor='registration-username-input'>
                        Choose a username<Required />
                    </Label>
                    <Input
                        id='registration-username-input'
                        name='username'
                        required
                    />
                </div>
                <div>
                    <Label htmlFor='registration-password-input'>
                        Choose a password<Required />
                    </Label>
                    <Input
                        id='registration-password-input'
                        name='password'
                        type='password'
                        required
                    />
                </div>
                <div>
                    <Button type="submit">
                        Sign up
                        </Button>
                </div>
                {' '}
                <Link className="already" to='/login'>Already have an account?</Link>
                <Description />
                </form>
            </section >
        )
    }
}

