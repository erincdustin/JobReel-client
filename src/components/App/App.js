import React from 'react'
import { Route, Switch } from 'react-router-dom'
import PrivateRoute from '../PrivateRoute/PrivateRoute'
import PublicOnlyRoute from '../PublicOnlyRoute/PublicOnlyRoute'
import RegistrationRoute from '../../routes/RegistrationRoute/RegistrationRoute'
import './App.css';
import Header from '../Header/Header'
import Footer from '../Footer/Footer'
import LoginRoute from '../../routes/LoginRoute/LoginRoute'
import DashboardRoute from '../../routes/DashboardRoute/DashboardRoute';
import SavedJobsRoute from '../../routes/SavedJobsRoute/SavedJobsRoute';
import SavedEventsRoute from '../../routes/SavedEventsRoute/SavedEventsRoute';
import SavedCompaniesRoute from '../../routes/SavedCompaniesRoute/SavedCompaniesRoute';
import JobsList from '../JobsList/JobsList';
import JobSearchForm from '../JobSearchForm/JobSearchForm';
import Meetup from '../Meetup/Meetup';

export default function App() {
  return (
    <div className='App'>
      <Header />
      <main>
        <Switch>
          <Route
            exact path={'/meetups'}
            component={Meetup}
          />
          <PublicOnlyRoute
            exact path={'/'}
            component={RegistrationRoute}
          />
          <PublicOnlyRoute
            path={'/login'}
            component={LoginRoute}
          />
          <PrivateRoute
            path={'/dashboard'}
            component={DashboardRoute}
          />
          <PrivateRoute
            path={'/saved-jobs'}
            component={SavedJobsRoute}
          />
          <PrivateRoute
            path={'/saved-events'}
            component={SavedEventsRoute}
          />
          <PrivateRoute
            path={'/jobs'}
            component={JobSearchForm}
          />
          <PrivateRoute
            exact path={'/JobsList'}
            component={JobsList}
          />
          <PrivateRoute
            path={'/companies'}
            component={SavedCompaniesRoute}
          />
        </Switch>
      </main>
      <Footer />
    </div>
  );
}