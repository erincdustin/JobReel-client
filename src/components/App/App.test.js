import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import App from './App';
import { BrowserRouter } from 'react-router-dom'
import { JobReelProvider } from '../../context/JobReelContext'
import RegistrationRoute from '../../routes/RegistrationRoute/RegistrationRoute';


describe('App smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<JobReelProvider><BrowserRouter><App /></BrowserRouter></JobReelProvider>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})

describe('App snapshot test', () => {
  var focusOnTargetSpy = jest.fn()

  jest
    .spyOn(RegistrationRoute.prototype, 'componentDidMount')
    .mockImplementation(focusOnTargetSpy)

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<JobReelProvider><BrowserRouter><App /></BrowserRouter></JobReelProvider>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(focusOnTargetSpy).toHaveBeenCalled()
  })
})