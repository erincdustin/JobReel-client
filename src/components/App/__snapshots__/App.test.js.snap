// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test renders the UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <img
          alt="jobreel-logo"
          src="jobreellogo200.png"
        />
      </a>
    </div>
    <div
      className="nav"
    >
      <nav
        role="navigation"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
         
        <a
          href="/"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </nav>
    </div>
  </header>
  <main>
    <div
      className="registration-page"
    >
      <div
        className="registration-image"
      >
        <img
          alt="registration-background"
          src="registration.svg"
        />
      </div>
      <section
        className="app-description"
      >
        <h1>
          JobReel
        </h1>
        <h2>
          Cast a wide net for your dream job.
        </h2>
        <p>
          There are a lot of moving pieces in a job hunt. Keep yourself organized and on task with JobReel's all-encompassing interface.  Search new postings and expand your network with our integrated job and industry event search, and track your application process from start to finish through the dashboard.
        </p>
      </section>
      <div
        className="form-registration"
      >
        <form
          id="form-container"
          onSubmit={[Function]}
        >
          <div
            className="error-message"
            role="alert"
          />
          <div>
            <label
              className="Label"
              htmlFor="registration-email-input"
            >
              Email:
              <span
                className="Required"
              >
                *
              </span>
            </label>
            <br />
            <input
              className="Input"
              id="registration-email-input"
              name="email"
              required={true}
              type="text"
            />
          </div>
          <div>
            <label
              className="Label"
              htmlFor="registration-first-name-input"
            >
              First name:
              <span
                className="Required"
              >
                *
              </span>
            </label>
            <br />
            <input
              className="Input"
              id="registration-first-name-input"
              name="first_name"
              required={true}
              type="text"
            />
          </div>
          <div>
            <label
              className="Label"
              htmlFor="registration-last-name-input"
            >
              Last name:
              <span
                className="Required"
              >
                *
              </span>
            </label>
            <br />
            <input
              className="Input"
              id="registration-last-name-input"
              name="last_name"
              required={true}
              type="text"
            />
          </div>
          <div>
            <label
              className="Label"
              htmlFor="registration-username-input"
            >
              Choose a username
              <span
                className="Required"
              >
                *
              </span>
            </label>
            <br />
            <input
              className="Input"
              id="registration-username-input"
              name="username"
              required={true}
              type="text"
            />
          </div>
          <div>
            <label
              className="Label"
              htmlFor="registration-password-input"
            >
              Choose a password
              <span
                className="Required"
              >
                *
              </span>
            </label>
            <br />
            <input
              className="Input"
              id="registration-password-input"
              name="password"
              required={true}
              type="password"
            />
          </div>
          <br />
          <div>
            <button
              className="Button"
              type="submit"
            >
              Sign up
            </button>
          </div>
           
          <br />
          <a
            className="already"
            href="/login"
            onClick={[Function]}
          >
            Already have an account?
          </a>
        </form>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <div
      className="team-info"
    >
      'Some Team Information goes here'
    </div>
    <div
      className="links-to-tech"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        GitHub Client
      </a>
      <br />
      <a
        href="/"
        onClick={[Function]}
      >
        GitHub Server
      </a>
    </div>
  </footer>
</div>
`;
